ProjectBasic/app/src/main/java/com/example/moviedb/screen/detail/DetailFragment.kt

```diff
@@ -0,0 +1,118 @@
+package com.example.moviedb.screen.detail
+
+import android.content.ActivityNotFoundException
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import androidx.lifecycle.Observer
+import com.example.moviedb.R
+import com.example.moviedb.data.model.Genre
+import com.example.moviedb.data.model.Movie
+import com.example.moviedb.data.model.Trailer
+import com.example.moviedb.databinding.DetailFragmentBinding
+import com.example.moviedb.screen.MainActivity
+import com.example.moviedb.screen.base.BaseFragment
+import com.example.moviedb.screen.detail.adapter.CastAdapter
+import com.example.moviedb.screen.detail.adapter.TrailerAdapter
+import com.example.moviedb.screen.home.MovieAdapter
+import com.example.moviedb.utils.Constant
+import com.google.android.material.chip.Chip
+import kotlinx.android.synthetic.main.detail_fragment.*
+import kotlinx.android.synthetic.main.toolbar_base.*
+import org.koin.androidx.viewmodel.ext.android.viewModel
+
+class DetailFragment() : BaseFragment<DetailFragmentBinding, DetailViewModel>() {
+
+    override val viewModel: DetailViewModel by viewModel()
+
+    private val onClickListener = object : TrailerAdapter.OnClickListener {
+        override fun onItemClick(trailer: Trailer) {
+            trailer.key?.let { watchYoutubeVideo(it) }
+        }
+    }
+    private val viewModelCastAdapter = CastAdapter()
+    private val viewModelTrailerAdapter = TrailerAdapter(onClickListener)
+
+    override val layoutId: Int
+        get() = R.layout.detail_fragment
+
+    override fun initComponents(view: DetailFragmentBinding) {
+        movie?.let { viewModel.getDataMovieDetail(it) }
+        initToolBar()
+        addListener(view)
+    }
+
+    private fun addListener(view: DetailFragmentBinding) {
```

Comment:
đặt tên chung 1 format
```suggestion
    private fun registerViewModel(view: DetailFragmentBinding) {
```